File: app/src/main/java/com/dkarakaya/songify/YoutubeDataModel.java
package com.dkarakaya.songify;

import android.os.Parcelable;
import android.os.Parcel;

public class YoutubeDataModel implements Parcelable {
    private String title = "";
    private String channelTitle = "";
    private String publishedAt = "";
    private String thumbnail = "";
    private String video_id = "";

    public String getVideo_id() {
        return video_id;
    }

    public void setVideo_id(String video_id) {
        this.video_id = video_id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getChannelTitle() {
        return channelTitle;
    }

    public void setChannelTitle(String channelTitle) {
        this.channelTitle = channelTitle;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(channelTitle);
        dest.writeString(publishedAt);
        dest.writeString(thumbnail);
        dest.writeString(video_id);
    }

    public YoutubeDataModel() {
        super();
    }


    protected YoutubeDataModel(Parcel in) {
        this();
        readFromParcel(in);
    }

    public void readFromParcel(Parcel in) {
        this.title = in.readString();
        this.channelTitle = in.readString();
        this.publishedAt = in.readString();
        this.thumbnail = in.readString();
        this.video_id = in.readString();

    }

    public static final Creator<YoutubeDataModel> CREATOR = new Creator<YoutubeDataModel>() {
        @Override
        public YoutubeDataModel createFromParcel(Parcel in) {
            return new YoutubeDataModel(in);
        }

        @Override
        public YoutubeDataModel[] newArray(int size) {
            return new YoutubeDataModel[size];
        }
    };
}

File: app/src/main/java/com/dkarakaya/songify/YoutubeDataModel.kt
package com.dkarakaya.songify

import android.os.Parcel
import android.os.Parcelable

data class YoutubeDataModel(var title: String? = "",
                            var channelTitle: String? = "",
                            var publishedAt: String? = "",
                            var thumbnail: String? = "",
                            var videoId: String? = "") : Parcelable {


    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(channelTitle)
        dest.writeString(publishedAt)
        dest.writeString(thumbnail)
        dest.writeString(videoId)
    }

    private constructor(parcel: Parcel) : this() {
        readFromParcel(parcel)
    }

    private fun readFromParcel(parcel: Parcel) {
        title = parcel.readString()
        channelTitle = parcel.readString()
        publishedAt = parcel.readString()
        thumbnail = parcel.readString()
        videoId = parcel.readString()
    }

    companion object {
        val CREATOR: Parcelable.Creator<YoutubeDataModel> = object : Parcelable.Creator<YoutubeDataModel> {
            override fun createFromParcel(parcel: Parcel): YoutubeDataModel? {
                return YoutubeDataModel(parcel)
            }

            override fun newArray(size: Int): Array<YoutubeDataModel?> {
                return arrayOfNulls(size)
            }
        }
    }
}
